iib/workers/tasks/opm_operations.py

```diff
@@ -671,33 +670,35 @@ def opm_generate_dockerfile(
         )
         return dockerfile_path
 
-    cmd = [
-        Opm.opm_version,
-        'generate',
-        'dockerfile',
-        os.path.abspath(fbc_dir),
-        '--binary-image',
-        binary_image,
-    ]
-
-    log.info('Generating Dockerfile with binary image %s' % binary_image)
-    run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to generate Dockerfile for file-based catalog')
-
-    # check if opm command generated Dockerfile successfully
+    log.info('Creating Dockerfile with binary image %s' % binary_image)
     dockerfile_path_opm_default = os.path.join(base_dir, dockerfile_name_opm_default)
-    if not os.path.isfile(dockerfile_path_opm_default):
-        error_msg = f"Cannot find generated Dockerfile at {dockerfile_path_opm_default}"
-        log.error(error_msg)
-        raise IIBError(error_msg)
+    with open(dockerfile_path_opm_default, 'w') as dockerfile:
+        dockerfile.write(
+            textwrap.dedent(
+                f"""\
+                    FROM {binary_image}
+
+                    # Configure the entrypoint and command
+                    ENTRYPOINT ["/bin/opm"]
+                    CMD ["serve", "/configs", "--cache-dir=/tmp/cache"]
+
+                    # Copy declarative config root and cache into image
+                    ADD catalog /configs
+                    COPY --chown=1001:0 cache /tmp/cache
+
+                    # Set DC-specific label for the location of the DC root directory
+                    # in the image
+                    LABEL operators.operatorframework.io.index.configs.v1=/configs
+                """
```

Comment:
Why not add also the remaining three commands? (I am okay with the current state, if there is a reason why not to add those).

Something like:
`ADD database/index.db /var/lib/iib/_hidden/do.not.edit.db
LABEL com.redhat.index.delivery.version=\"{delivery_version}\"
LABEL com.redhat.index.delivery.distribution_scope=\"{distribution_scope}\"
`

Reply:
I had a similar idea so we can remove all the functions which modify Dockerfile and have just one that generates it all in one call. 
However if it is a lot of work we can have different jira for that.

Reply:
yes! what Jan said. It'll be major restructuring. Let's have a follow-up Jira to do that